QUARKUS-4221.md

```diff
@@ -0,0 +1,75 @@
+# QUARKUS-4221 Productization of Picocli extension
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-4221
+
+PR: https://github.com/quarkus-qe/quarkus-test-suite/pull/2003
+## Scope of the testing
+### What to test
+
+The test coverage will be verified for environments: JVM for Baremetal and Openshift.
+
+* Verify simple command line application with some annotations such as: `@CommandLine.Command` / `@CommandLine.Option` 
```

Comment:
Is there a coverage in upstream?

Reply:
Upstream unit tests coverage is here:
https://github.com/quarkusio/quarkus/tree/main/integration-tests/picocli

Reply:
What scenarios a covered here? Is there some, which are already done upstream and so we do not need to add it in our TS?

Reply:
Yes, there are some basic tests upstream related to commands that I also use here, but in a slightly different way. In my tests, I use a @QuarkusApplication with a RestService running, so they are tested from a different angle. However, if you believe those upstream tests cover the scenarios well enough and my tests are redundant, I'll remove them.

Reply:
@QuarkusApplication and RestService are only used by Quarkus QE, so they aren't of interest to anyone else :-)

It is a task for the author of TP do look into the existing coverage and decide, whether it is enough or not, and if it isn't — add the lacking scenarios. Reviewers task is to check, whether author do not forget about it. Ideally, the current coverage should be described, so anyone who works on this in the future will be able to find our reasoning.

Reply:
IMHO these tests extend beyond upstream unit tests by utilizing @QuarkusApplication and @RestService (as a bunch of ours test in several modules), this allows us to perform some kind of integration testing, so I'll keep them and I've described it in the TP.